### ISSUE TITLE
Add `flush_bytes` configuration

### ISSUE THREAD

**Issue by criccomini** [slatedb/slatedb#30]
SlateDB currently flushes mem_tables on a fixed interval defined by `flush_ms`. This is fine for a first pass, but I think we should also flush if a maximum byte size is reached. This is very similar to how Kafka's [`linger.ms` and `batch.size`](https://www.geeksforgeeks.org/apache-kafka-linger-ms-and-batch-size/) producer settings work.

For high write throughput (or bursty) use cases, the memtable could get quite large if we simply wait some number of millseconds. This could cause a long delay on upload to object storage. By short-circuiting the flush_ms wait time when a memtable has reached a size threshold, we can freeze the memtable and begin the upload immediately.

Note that doing this could result in parallel SST uploads since we now have multiple threads that could trigger a flush. Parallel uploads mean we might expose out of order reads if an SST A completes its upload *after* SST B. I *think* this is OK and still compatible with snapshot isolation since a writer will block on its own write before reading? In the case of another reader that reads SST B before SST A is written, any keys in SST B that are also in A would be overwritten, so I think it's also safe? This might need some thought...

**Comment by criccomini** [comment]
I'm going to close this. `max_unflushed_memtable` and `l0_sst_size_bytes` let us control this now. Suppose we set `l0_sst_size_bytes` to 64MiB. As soon as the memtable hits >=  64MiB, it'll be flushed. As soon as 64MiB _more_ writes come in, a second memtable flush is triggered. If 64MiB _more_ bytes come in, the writer will be blocked after this until at least one of the memtables flushes.

There's still the pathological case where a single k/v pair is _much_ bigger than `l0_sst_size_bytes`, which would cause memory pressure. But that's always going to be true since the data is big.

So yea, I'm closing this as won't fix.

**Comment by criccomini** [comment]
Re-opening this. I think it's worth adding. I'm not as concerned about the stuff in #31 anymore.

**Comment by criccomini** [comment]
This is is done as part of #519. It now has `max_wal_bytes_size`, which uses the `l0_max_size_bytes` config.
